prow/cmd/webhook-server/main.go

```diff
@@ -178,6 +179,8 @@ func main() {
 	defer interrupts.WaitForGracefulShutdown()
 	mux := http.NewServeMux()
 	mux.HandleFunc("/validate", wa.serveValidate)
+	health := pjutil.NewHealthOnPort(8081)
```

Comment:
Please start serving this earlier. The idea is to serve the liveness endpoint early during startup and the readiness endpoint once startup is complete.
You should be able to copy and paste much of this from other components. For example the `prow-controller-manager`: https://github.com/kubernetes/test-infra/blob/4f214fc0d0aa2beceffb0b070af6b42d06b5b163/prow/cmd/prow-controller-manager/main.go#L119-L121
(Notice that `WaitForGracefulShutdown` is defered early as well since `NewHealth` uses the interrupts package.)

Rather than using `8081` as a "magic number" here please use the following to always serve on the default port: https://github.com/kubernetes/test-infra/blob/4f214fc0d0aa2beceffb0b070af6b42d06b5b163/prow/pjutil/health.go#L36-L38
or use the `instrumentationOptions` flagset like in `prow-controller-manager`.